godtools/App/Features/Dashboard/Presentation/Tools/ToolsViewModel.swift

```diff
@@ -60,11 +60,32 @@ class ToolsViewModel: ObservableObject {
         
         showsFavoritingToolBanner = !favoritingToolMessageCache.favoritingToolMessageDisabled
         
-        let anyCategorySelection = getToolFilterCategoriesUseCase.getAnyCategoryDomainModel()
         let anyLanguageSelection = getToolFilterLanguagesUseCase.getAnyLanguageFilterDomainModel()
+        let anyCategorySelection = getToolFilterCategoriesUseCase.getAnyCategoryDomainModel()
         toolFilterLanguageSelectionPublisher = CurrentValueSubject(anyLanguageSelection)
         toolFilterCategorySelectionPublisher = CurrentValueSubject(anyCategorySelection)
         
+        getUserFiltersUseCase.getUserFiltersPublisher()
+            .flatMap { userFilters in
+                
+                return Publishers.CombineLatest(
+                    getToolFilterCategoriesUseCase.getCategoryFilterPublisher(with: userFilters.categoryFilterId),
+                    getToolFilterLanguagesUseCase.getLanguageFilterPublisher(from: userFilters.languageFilterId)
+                )
+                .eraseToAnyPublisher()
+            }
+            .sink { categoryFilter, languageFilter in
+                
+                if let categoryFilter = categoryFilter {
+                    self.toolFilterCategorySelectionPublisher.send(categoryFilter)
+                }
+                
+                if let languageFilter = languageFilter {
+                    self.toolFilterLanguageSelectionPublisher.send(languageFilter)
+                }
+            }
+            .store(in: &cancellables)
```

Comment:
This is another problem.  The `UserFilterDomainModel` returns filter IDs only, not the actual objects.  So here I'm fetching the object once I get the IDs.  I was inclined to put this logic directly into `GetUserFiltersUseCase`, but that would mean another use case inside a use case.

Reply:
Hey @rachaelblue let's come back and revisit this one.  I'm going to work on updating the data layer to hopefully make getting the number of tools available a bit easier to query.  Then it would be good to re-visit some of the UseCases in this feature to see how we can update them for using dependency inversion and writing behavior tests too.